packages/pwa-kit-react-sdk/src/ssr/universal/components/loading/index.jsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import {useHistory, useLocation} from 'react-router-dom'
+import {useQueryClient} from '@tanstack/react-query'
+
+const Loading = () => {
+    const history = useHistory()
+    const location = useLocation()
+
+    // TODO: can we assume react-query is in use?
+    const queryClient = useQueryClient()
+    queryClient.invalidateQueries()
```

Comment:
🤔 maybe this should be run once only.. maybe I'll move it to inside the `useEffect` below.